packages/ann/ann/test/test-sparse.lua

```diff
@@ -41,3 +41,58 @@ T("SparseDotProductTest",
       end
     end
 end)
+--
+T("SparseLogistic",
```

Comment:
I have added this test, which uses finite differences to check the gradients computed by sparse logistic component.